src/RedisSessionStateProvider/RedisSessionStateProvider.cs

```diff
@@ -199,6 +199,10 @@ private SessionStateStoreData GetItemFromSessionStore(bool isWriteLockRequired,
                 lockAge = TimeSpan.Zero;
                 lockId = 0;
                 actions = SessionStateActions.None;
+                if (id == null)
```

Comment:
Can we have some explanation about why session id is null? Also, please add some test cases